mozilla-release/mobile/android/app/src/bond/java/org/mozilla/gecko/controlcenter/DashboardAdapter.java

```diff
@@ -0,0 +1,243 @@
+                        viewHolder.itemMoneyBarValueView.setText(String.valueOf(progress +
+                                MIN_MONEY_BAR_VALUE).concat(" €"));
+                    }
+
+                    @Override
+                    public void onStartTrackingTouch(SeekBar seekBar) {
+
+                    }
+
+                    @Override
+                    public void onStopTrackingTouch(SeekBar seekBar) {
+
+                    }
+                });
+
+                viewHolder.itemMoneyBarValueView.setVisibility(View.VISIBLE);
+            }
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return position == 0 ? ItemType.TWO_VIEWS.ordinal() : ItemType.ONE_VIEW.ordinal();
+    }
+
+    @Override
+    public int getItemCount() { // -1, because first 2 items combined in the first view
+        return mDashboardItems.size() - 1;
+    }
+
+    void setData(GeckoBundle data) {
+        // @TODO fetch the data and fill the list
+    }
+
+    void addItems(List<DashboardItemEntity> items) {
+        mDashboardItems.clear();
+        mDashboardItems.addAll(items);
+        notifyDataSetChanged();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
+        public ImageView itemIconView;
+        public TextView itemMeasurementView;
+        public TextView itemTitleView;
+        public TextView itemContentView;
+        public SeekBar itemMoneyBarView;
+        public TextView itemMoneyBarValueView;
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            itemIconView = (ImageView) itemView.findViewById(R.id.item_icon);
+            itemMeasurementView = (TextView) itemView.findViewById(R.id.item_measurement);
+            itemTitleView = (TextView) itemView.findViewById(R.id.item_title);
+            itemContentView = (TextView) itemView.findViewById(R.id.item_content);
+            itemMoneyBarView = (SeekBar) itemView.findViewById(R.id.item_money_bar);
+            itemMoneyBarValueView = (TextView) itemView.findViewById(R.id.item_money_bar_value);
+        }
+    }
+
+    public class ItemStructure {
+        public ImageView itemIconView;
+        public TextView itemMeasurementView;
+        public TextView itemTitleView;
+        public TextView itemContentView;
+    }
+
+    public class ViewHolder2 extends RecyclerView.ViewHolder {
+        final ItemStructure leftItem, rightItem;
+
+        public ViewHolder2(View itemView) {
+            super(itemView);
+            leftItem = new ItemStructure();
+            final View leftView = itemView.findViewById(R.id.left_item);
+            findViews(leftView, leftItem);
+            rightItem = new ItemStructure();
+            final View rightView = itemView.findViewById(R.id.right_item);
+            findViews(rightView, rightItem);
+        }
+
+        public void findViews(View view, ItemStructure item) {
```

Comment:
This should return a properly constructed `ItemStructure`, instead of filling one.
For the future, please mark parameters as `outParamName` when you have a parameter that get changed inside the method.